remote-api/src/main/java/org/alfresco/rest/api/nodes/NodesEntityResource.java

```diff
@@ -225,5 +284,102 @@ public DirectAccessUrl requestContentDirectUrl(String nodeId, DirectAccessUrlReq
         }
         return directAccessUrl;
     }
+
+    /**
+     * Folder Size - returns size of a folder.
+     *
+     * @param nodeId String id of folder - will also accept well-known alias, eg. -root- or -my- or -shared-
+     *               Please refer to OpenAPI spec for more details !
+     * Returns the response message i.e. Request has been acknowledged with 202
+     * GET/size endpoint to check if the action's status has been completed, comprising the size of the node in bytes.
+     *               <p>
+     *               If nodeId does not represent a folder, InvalidNodeTypeException (status 422).
+     */
+    @Operation("calculate-folder-size")
+    @WebApiDescription(title = "Calculating Folder Size", description = "Calculating size of a folder", successStatus = Status.STATUS_ACCEPTED)
+    @WebApiParameters({@WebApiParam(name = "nodeId", title = "The unique id of Execution Job", description = "A single nodeId")})
+    public Map<String, Object> calculateFolderSize(String nodeId, Void ignore, Parameters parameters, WithResponse withResponse)
+    {
+        NodeRef nodeRef = nodes.validateNode(nodeId);
+        QName qName = nodeService.getType(nodeRef);
+        Map<String, Object> result = new HashMap<>();
+        validatePermissions(nodeRef, nodeId);
+
+        if(!FOLDER.equalsIgnoreCase(qName.getLocalName()))
+        {
+            throw new InvalidNodeTypeException(INVALID_NODEID);
+        }
+
+        try
+        {
+            FolderSizeImpl folderSizeImpl = new FolderSizeImpl(actionService);
+            return folderSizeImpl.executingAsynchronousFolderAction(nodeRef, defaultItems, result, simpleCache);
+        }
+        catch (Exception alfrescoRuntimeError)
+        {
+            LOG.error("Exception occurred in NodesEntityResource:calculateFolderSize {}", alfrescoRuntimeError.getMessage());
+            throw new AlfrescoRuntimeException("Exception occurred in NodesEntityResource:calculateFolderSize",alfrescoRuntimeError);
+        }
+    }
+
+    @Override
+    @WebApiDescription(title = "Returns Folder Node Size", description = "Returning a Folder Node Size")
+    @WebApiParameters({@WebApiParam(name = "nodeId", title = "The unique id of Execution Job", description = "A single nodeId")})
+    @BinaryProperties({"size"})
```

Comment:
What is the purpose of `BinaryProperties`? You're not returning binary content to the client.